AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClientOperations/AWSMobileClient+Password.swift

```diff
@@ -0,0 +1,86 @@
+//
+// Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+import Foundation
+
+extension AWSMobileClient {
+
+    /// When called, will send a confirmation code to user which can be used to reset the password.
+    ///
+    /// - Parameters:
+    ///   - username: username of the user who forgot the password.
+    ///   - clientMetaData: A map of custom key-value pairs that you can provide as input for any
+    ///   custom workflows that this action triggers.
+    ///   - completionHandler: completionHandler which will be called when result is available.
+    public func forgotPassword(username: String,
+                               clientMetaData: [String:String] = [:],
+                               completionHandler: @escaping ((ForgotPasswordResult?, Error?) -> Void)) {
+        let user = self.userPoolClient?.getUser(username)
+        user!.forgotPassword(clientMetaData).continueWith { (task) -> Any? in
```

Comment:
What about unwrapping with a guard and using `fatalError` to provide a useful message in case the value is ever nil? Or is it only nil because `userPoolClient` is optional? What I would do then is guard unwrap that value first with the fatal error with useful message.

Reply:
Not changing this since this PR just moves the code to specific files and no behavior changes are made.